### PR TITLE
VIH-8245 witnesses displayed as Available when in the waiting room

### PR DESCRIPTION
**Pull request by alberto-hortiguela**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-8245

### Change description ###
To display "Available" tooltip message when a witness on the WR is callable

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```
image

### CODE REVIEW

**Comment by MichaelKeaneathmcts** [comment]
This might need an automation test at some point.  Is there a unique reference to the web element?  I'm looking for code such as <... id={participantId}-element-name ...>

Any automation tests can be brittle unless there is a very clear reference to an element, searching the DOM can mean that the next web change breaks the tests.

**Comment by FarronF** [comment]
> 
> 
> This might need an automation test at some point. Is there a unique reference to the web element? I'm looking for code such as <... id={participantId}-element-name ...>
> 
> Any automation tests can be brittle unless there is a very clear reference to an element, searching the DOM can mean that the next web change breaks the tests.

@MichaelKeaneathmcts Maybe we should discuss a way we could approach this in the future. I'm ensuring a component I'm working on now has ids as such `judge-context-menu-participant-${participantId}-${elementId}` but all of this is pretty adhoc and we don't have an easy way to generate these ids. And I don't think we have a clear style for what these ids should look like.

**Review by FarronF — CHANGES_REQUESTED** [review]
Looking at Lorenzo's comment on the ticket it states the following cases

    Joining
    Available
    In consultation
    Disconnected
    Not signed in

I don't see "In consultation" covered here. We should implement that as part of this.

**Comment by alberto-hortiguela** [comment]
> Looking at Lorenzo's comment on the ticket it states the following cases
> 
> ```
> Joining
> Available
> In consultation
> Disconnected
> Not signed in
> ```
> 
> I don't see "In consultation" covered here. We should implement that as part of this.

You are right. However should not be then a ticket ( with points ) instead of a bug fix? This looks like new functionality

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participants-panel/participants-panel.component.ts [hunk 1]
```diff
@@ -506,16 +506,27 @@ export class ParticipantsPanelComponent implements OnInit, OnDestroy {
...
+                participant.displayName + ': ' + this.getTranslatedText('participant-available') + this.getAdditionalText(participant);
         }
         if (participant.isDisconnected()) {
-            return participant.displayName + ': ' + this.getTranslatedText('disconnected') + this.getAdditionalText(participant);
+            toolTipText = participant.displayName + ': ' + this.getTranslatedText('disconnected') + this.getAdditionalText(participant);
         }
-        return participant.displayName + this.getAdditionalText(participant);
+
+        if (!toolTipText) {
+            toolTipText = participant.displayName + this.getAdditionalText(participant);
```
Comment (omio-ma) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participants-panel/participants-panel.component.ts:526]: we can use this to initialize the toolTipText property, removes the need for this if block

**Comment by FarronF** [comment]
> 
> 
> > Looking at Lorenzo's comment on the ticket it states the following cases
> > ```
> > Joining
> > Available
> > In consultation
> > Disconnected
> > Not signed in
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I don't see "In consultation" covered here. We should implement that as part of this.
> 
> Is there a way to have a hearing in session and participants having a private consultation?
> If we are bringing new functionality should we not create a ticket ( with points ) instead of a bug fix one?

Yeah I think you're right actually. This case either was missed as an AC or was not considered. Lets discuss this with the team tomorrow.

**Comment by omio-ma** [comment]
> This might need an automation test at some point. Is there a unique reference to the web element? I'm looking for code such as <... id={participantId}-element-name ...>
> 
> Any automation tests can be brittle unless there is a very clear reference to an element, searching the DOM can mean that the next web change breaks the tests.

I had to check the front end code for this, currently the tooltip is an angular directive, and it doesn't accept a property that can be used to dynamically create a unique id. It is do-able but I believe that should be part of another ticket as it would impact all the other tooltips we have in our application. Also I don't know how easily we can do an automation test for this, because this tooltip is actually not being hidden instead being pushed out off screen (ie, its using top and left css properties to take it off display) which would mean we cant just rely on asserting that the tooltip element is visible when hovered over the judge's name

**Comment by MichaelKeaneathmcts** [comment]
Thanks for the feedback, in this case then an ID is probably not required.

I hope that everyone will think about what elements can be referenced by automation tests going forward.

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participants-panel/participants-panel.component.ts (hunk 1 above)
Comment (alberto-hortiguela) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participants-panel/participants-panel.component.ts]: that's done now

**Review by omio-ma — APPROVED** [review]

**Review by krishnapottekkat-hmcts — APPROVED** [review]

**Review by FarronF — APPROVED** [review]

**Pull request merged**